File: Java/Tema4_Ejercicio4.java
/*
 * TEMA 4: ESTRUCTURAS REPETITIVAS - EJERCICIO 4
 * ===============================================
 * Escriba un programa que imprima un triángulo usando bucles.
 */

import java.util.Scanner;

public class Tema4_Ejercicio4 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Digite la altura del triángulo: ");
        int altura = scanner.nextInt();
        
        System.out.println("\n=== TRIÁNGULO ASCENDENTE ===");
        
        for (int i = 1; i <= altura; i++) {
            for (int j = 0; j < i; j++) {
                System.out.print("*");
            }
            System.out.println();
        }
        
        System.out.println("\n=== TRIÁNGULO DESCENDENTE ===");
        
        for (int i = altura; i >= 1; i--) {
            for (int j = 0; j < i; j++) {
                System.out.print("*");
            }
            System.out.println();
        }
        
        scanner.close();
    }
}

/*
 * SALIDA ESPERADA:
 * Digite la altura del triángulo: 4
 * 
 * === TRIÁNGULO ASCENDENTE ===
 * *
 * **
 * ***
 * ****
 * 
 * === TRIÁNGULO DESCENDENTE ===
 * ****
 * ***
 * **
 * *
 */

File: JS/Tema4_Ejercicio4.js
/*
 * TEMA 4: ESTRUCTURAS REPETITIVAS - EJERCICIO 4
 * ==============================================
 * Crear patrones triangulares.
 * 
 * EXPLICACIÓN:
 * - Triángulos ascendentes y descendentes
 * - Variación de bucles anidados
 * - Espacios y caracteres
 */

console.log("=== PATRONES TRIANGULARES ===");

// Triángulo ascendente
console.log("Triángulo ascendente:");
for (let i = 1; i <= 5; i++) {
    let fila = "";
    for (let j = 1; j <= i; j++) {
        fila += "* ";
    }
    console.log(fila);
}

// Triángulo descendente
console.log("\nTriángulo descendente:");
for (let i = 5; i >= 1; i--) {
    let fila = "";
    for (let j = 1; j <= i; j++) {
        fila += "* ";
    }
    console.log(fila);
}

// Triángulo de números
console.log("\nTriángulo de números:");
for (let i = 1; i <= 4; i++) {
    let fila = "";
    for (let j = 1; j <= i; j++) {
        fila += j + " ";
    }
    console.log(fila);
}

/*
 * SALIDA ESPERADA:
 * === PATRONES TRIANGULARES ===
 * Triángulo ascendente:
 * * 
 * * * 
 * * * * 
 * * * * * 
 * * * * * * 
 * 
 * Triángulo descendente:
 * * * * * * 
 * * * * * 
 * * * * 
 * * * 
 * * 
 * 
 * Triángulo de números:
 * 1 
 * 1 2 
 * 1 2 3 
 * 1 2 3 4 
 */
